spec/services/instance/user_invitation_service_spec.rb

```diff
@@ -193,9 +193,11 @@ def invite
         let(:all_users) { existing_users + new_users }
 
         it 'defaults to :normal for roles' do
-          result_new, _, result_existing =
+          new_invitations, _, new_instance_users =
             subject.send(:invite_users, temp_form_hash_from_attributes(all_users))
-          (result_new + result_existing).each do |invit|
+          new_invitations_and_instance_users = new_invitations + new_instance_users
```

Comment:
Since the `before_validation` callback is used now, the records have to be saved first before checking for default roles and invitation keys.